packages/pg-v5/test/lib/psql.js

```diff
@@ -286,44 +322,39 @@ describe('psql', () => {
             PGSSLMODE: 'prefer'
           })
 
-          spawnStub.callsFake((commandName, args, opts) => {
-            expect(commandName, 'to equal', 'psql')
-            expect(args, 'to equal', expectedArgs)
-            expect(opts.stdio, 'to equal', 'inherit')
-            Object.entries(expectedEnv).forEach(([key, expectedValue]) => {
-              expect(opts.env[key], 'to equal', expectedValue)
-            })
-
-            return {
-              on: (key, callback) => {
-                if (key === 'close') {
-                  callback(new Error(0))
-                }
-              }
+          const mock = mockSpawn(
+            'psql',
+            expectedArgs,
+            {
+              stdio: 'inherit',
+              env: expectedEnv
             }
+          )
+
+          mock.callsFake(() => {
+            fakePsqlProcess.start()
+            return fakePsqlProcess
           })
 
-          return psql.interactive(db)
-            .finally(() => {
-              spawnStub.restore()
-              existsSyncStub.restore()
-              statSyncStub.restore()
-            })
+          const promise = psql.interactive(db)
+          await fakePsqlProcess.waitForStart()
+          await fakePsqlProcess.simulateExit(0)
+          mock.verify()
+          await ensureFinished(promise)
         })
       })
 
       context('when HEROKU_PSQL_HISTORY is a valid file path', () => {
-        it('is the path to the history file', () => {
-          const spawnStub = sinon.stub(require('child_process'), 'spawn')
-
-          const existsSyncStub = sinon
-            .stub(require('fs'), 'existsSync')
-            .callsFake(() => true)
+        beforeEach(function () {
```

Comment:
i refactored the tests here to use a temporary file/path; it was easier than trying to anticipate or create a mock for any call to `fs` possible (a lot of third party libraries were using `fs`, for example). These temporary files get cleaned up automatically thanks to the `tmp` library, but I've added cleanup code in the `afterEach` just for good hygiene if copy/pasta happens later.